lambda_src/drivers/process_https.py

```diff
@@ -98,10 +98,23 @@ def process_row(
     req_method: str = method.upper()
 
     if json:
+        
         req_data: Optional[bytes] = (
             json if json.startswith('{') else dumps(parse_header_dict(json))
         ).encode()
         req_headers['Content-Type'] = 'application/json'
+        if json.startswith('arn'):
+            data=(decrypt_if_encrypted(json))
+            data=loads(data)
+            json_host=data['host']
+            if json_host and req_host and json_host != req_host:
```

Comment:
I think it's better to include `json-body` as an auth option in the part that already has this check, could you add it there and re-use that structure rather than including a second one?